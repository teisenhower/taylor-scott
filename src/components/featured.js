import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'
import InternalLayout from '../layouts/internalLayout'
import style from './featured.module.css'

export const Featured = () => {
  const data = useStaticQuery(graphql`
    query {
      allFile(filter: { relativeDirectory: { eq: "images/featured" } }) {
        nodes {
          childImageSharp {
            fluid {
              ...GatsbyImageSharpFluid_withWebp
            }
          }
          name
        }
      }
    }
  `)
  return <FeaturedSection data={data} />
}
export const FeaturedSection = ({ data }) => {
  const imageGridNames = ['main', 'secondary', 'subA', 'subB']
  const { nodes: images } = data.allFile
  return (
    <InternalLayout id={style.featured}>
      <h3 className="sectionHeader upper">Featured Session</h3>
      <div className={style.featuredImages}>
        {images.map((image, index) => (
          <Img
            alt={image.name}
            key={image.name}
            className={style[imageGridNames[index]]}
            fluid={image.childImageSharp.fluid}
          />
        ))}
      </div>
      <div className={style.description}>
        <h4 className="upper">The Dale Family</h4>
        <p>
          Lorem ipsum dolor sit amet consectetur, adipisicing elit. Accusamus
          aliquam adipisci ducimus voluptate quia beatae quis ex, amet
          perferendis accusantium quisquam! Delectus libero excepturi
          consequatur fuga amet enim sapiente placeat quas quaerat cum
          architecto soluta impedit at exercitationem magnam esse doloremque
          repellendus voluptatum, cupiditate doloribus a qui dolorum eligendi.
          Consequuntur.
        </p>
        <button type="button" className="">
          View Session
        </button>
      </div>
    </InternalLayout>
  )
}
