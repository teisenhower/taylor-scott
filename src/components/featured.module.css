@value med from '../assets/breakpoints.module.css';

#featured {
  background: var(--grey);
  display: grid;
  align-items: center;
}
button {
  padding: 0 3.2rem;
  margin: 4rem 0 2.4rem 0;
  font-size: 1.2rem;
  line-height: 2.4rem;
  border: 2px solid var(--gold);
  justify-self: center;
}

.featuredImages {
  display: grid;
  gap: 1.6rem;
  grid-template-areas:
    'main main secondary secondary'
    'main main subA subB'
    'main main subA subB';
}

.main {
  grid-area: main;
}
.secondary {
  grid-area: secondary;
}
.subA {
  grid-area: subA;
}
.subB {
  grid-area: subB;
}
.description {
  height: 100%;
  display: grid;
  grid-template-rows: minmax(10px, max-content);
}

@media (med) {
  #featured {
    padding-top: 5.6rem;
    padding-bottom: 5.6rem;
    grid-template-columns: 75% 25%;
    grid-template-areas:
      'header header'
      'images description';
  }
  .section h3 {
    grid-area: header;
  }
  .featuredImages {
    padding-right: 2.4rem;
    grid-area: images;
  }
  .description {
    grid-area: description;
  }
  .description h4 {
    align-self: start;
    line-height: 1;
    padding-top: 0;
  }
  .description p {
    align-self: start;
    font-size: 1.4rem;
    line-height: 2.4rem;
  }
  .description button {
    font-size: 1.4rem;
    line-height: 3.2rem;
    justify-self: flex-start;
    align-self: flex-end;
    margin-bottom: 0;
  }
}
