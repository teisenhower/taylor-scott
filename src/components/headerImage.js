import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'
import style from './headerImage.module.css'

export const HeaderImage = () => {
  const data = useStaticQuery(graphql`
    query {
      fixedImage: file(relativePath: { eq: "images/header.jpg" }) {
        childImageSharp {
          fluid(quality: 100) {
            ...GatsbyImageSharpFluid_withWebp
          }
        }
      }
    }
  `)
  return <HeaderImg data={data} />
}

export const HeaderImg = ({ data }) => {
  return (
    <div>
      <Img
        className={style.headerImage}
        fluid={data.fixedImage.childImageSharp.fluid}
      />
    </div>
  )
}

export default HeaderImage
