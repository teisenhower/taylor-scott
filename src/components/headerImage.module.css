@value med from '../assets/breakpoints.module.css';

.headerImage {
  opacity: 0.6;
  margin: 1.6rem 1.6rem 0 1.6rem;
}

@media (med) {
  .headerImage {
    margin: 2.4rem 22rem 0 22rem;
  }
}
