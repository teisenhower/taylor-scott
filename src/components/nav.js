import { useStaticQuery, graphql } from 'gatsby'
import React, { useState, useEffect } from 'react'
import style from './nav.module.css'

export const Nav = () => {
  const data = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            navLinks
          }
        }
      }
    `,
  )
  return <NavComponent data={data} />
}

export const NavComponent = ({ data }) => {
  const [navActive, setNavActive] = useState(false)
  const [navPOS, setNavPOS] = useState()
  const [windowOrientation, setWindowOrientation] = useState()

  const toggleNavActive = () => {
    setNavActive(!navActive)
  }
  window.onorientationchange = e => {
    setTimeout(() => {
      setWindowOrientation(e.target.screen.orientation.angle)
    }, 100)
  }
  useEffect(() => {
    const overallNavHeight = document
      .getElementById('nav')
      .getBoundingClientRect().height
    const hamburgerIconHeight = document
      .querySelector('nav div')
      .getBoundingClientRect().height
    const POS = (overallNavHeight - hamburgerIconHeight) * -1
    setNavPOS(POS)
  }, [windowOrientation])
  return (
    <nav id="nav" style={navActive ? { bottom: '0' } : { bottom: navPOS }}>
      <ul>
        <div
          role="button"
          tabIndex="0"
          id={style.hamburger}
          onClick={toggleNavActive}
          onKeyDown={toggleNavActive}
        >
          X
        </div>
        {data.site.siteMetadata.navLinks.map(navItem => (
          <li key={navItem}>{navItem}</li>
        ))}
      </ul>
    </nav>
  )
}

export default Nav
