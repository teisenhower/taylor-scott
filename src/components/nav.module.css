@value med from '../assets/breakpoints.module.css';
nav {
  position: fixed;
  left: 0;
  background-color: #f7f7f7;
  width: 100vw;
  transition: all ease-in-out 0.25s;
  animation: fadein 1s;
  z-index: 1000;
}
ul {
  list-style: none;
  padding: 0;
  margin: 1.6rem;
  text-align: right;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}
li {
  font-size: 1.2rem;
}
#hamburger {
  padding-bottom: 1.6rem;
  width: 1.6rem;
  border: none;
  outline: none;
}
#hamburger:focus-visible {
  text-decoration: none;
  outline: 1px solid rgba(0, 0, 0, 0.5);
}
@media (med) {
  nav {
    top: 0;
    position: relative;
    width: 100%;
    background-color: unset;
    animation: none;
  }
  ul {
    display: block;
    margin: 0;
    padding-right: 22rem;
  }
  li {
    display: inline;
    padding-left: 5.2rem;
  }
  #hamburger {
    display: none;
  }
}

@keyframes fadein {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
