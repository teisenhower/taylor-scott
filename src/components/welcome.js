import React from 'react'
import style from './welcome.module.css'
import InternalLayout from '../layouts/internalLayout'

export const Welcome = () => (
  <InternalLayout id={style.welcome}>
    <h2>Hello &amp; Welcome</h2>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Vel magnam quia
      debitis, dolore ipsam dignissimos alias reprehenderit dolores odit
      tempore. Recusandae, alias? Perspiciatis enim aliquam architecto nisi.
      Quae, pariatur error amet recusandae unde incidunt. Eum, eius! Fugiat
      ullam enim amet voluptas et ad, quas nostrum repellendus corrupti hic nam
      explicabo quasi molestias ducimus ex eos pariatur quis, sit temporibus vel
      modi! Laudantium reiciendis, facilis corporis delectus recusandae quidem
      eligendi nesciunt voluptatibus nemo, eius magnam incidunt!
    </p>
  </InternalLayout>
)

export default Welcome
