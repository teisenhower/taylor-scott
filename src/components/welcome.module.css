@value med from '../assets/breakpoints.module.css';

#welcome {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 5.6rem;
}

#welcome > h2 {
  font-family: 'Montserrat', sans-serif;
  font-weight: 500;
  text-transform: uppercase;
  letter-spacing: 4px;
  margin-top: 4.8rem;
  padding-bottom: 0.8rem;
  font-size: 1.8rem;
  line-height: 2.4rem;
  border-bottom: 1px solid var(--gold);
}
#welcome > p {
  font-size: 1.2rem;
  text-align: center;
  line-height: 1.8rem;
  margin-top: 1.2rem;
}

@media (med) {
  #welcome > h2 {
    font-size: 2.4rem;
  }
  #welcome > p {
    font-size: 1.6rem;
    line-height: 2.4rem;
    padding-top: 1.6rem;
  }
}
