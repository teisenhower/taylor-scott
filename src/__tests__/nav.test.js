import React from 'react'
import renderer from 'react-test-renderer'
import { NavComponent } from '../components/nav'

it('renders correctly', () => {
  const data = {
    site: {
      siteMetadata: {
        navLinks: ['About', 'Pricing', 'Wedding', 'Blog', 'Contact'],
      },
    },
  }
  const component = renderer.create(<NavComponent data={data} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
