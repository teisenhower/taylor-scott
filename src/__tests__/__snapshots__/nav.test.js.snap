// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  id="nav"
  style={
    Object {
      "bottom": undefined,
    }
  }
>
  <ul>
    <div
      id="hamburger"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      X
    </div>
    <li>
      About
    </li>
    <li>
      Pricing
    </li>
    <li>
      Wedding
    </li>
    <li>
      Blog
    </li>
    <li>
      Contact
    </li>
  </ul>
</nav>
`;
